tests/test_manager.py

```diff
@@ -0,0 +1,110 @@
+from __future__ import absolute_import
+from __future__ import unicode_literals
+
+from __future__ import print_function
+from django.db.models.expressions import F
+from django.db.models.query import QuerySet
+from django.test import TestCase
+
+from django_cte import With, CTEQuerySet, CTEManager
+
+from .models import (
+    Order,
+    OrderFromLT40,
+    OrderLT40AsManager,
+    OrderCustomManagerNQuery,
+    OrderCustomManager,
+    LT40QuerySet,
+    LTManager,
+    LT25QuerySet,
+)
+
+
+class TestCTE(TestCase):
+    def test_cte_queryset_correct_defaultmanager(self):
+        self.assertEqual(type(Order._default_manager), CTEManager)
+        self.assertEqual(type(Order.objects.all()), CTEQuerySet)
+
+    def test_cte_queryset_correct_from_queryset(self):
+        self.assertEqual(type(OrderFromLT40.objects.all()), LT40QuerySet)
+
+    def test_cte_queryset_correct_queryset_as_manager(self):
+        self.assertEqual(type(OrderLT40AsManager.objects.all()), LT40QuerySet)
+
+    def test_cte_queryset_correct_manager_n_from_queryset(self):
+        self.assertTrue(isinstance(
+            OrderCustomManagerNQuery._default_manager, LTManager))
+        self.assertEqual(type(
+            OrderCustomManagerNQuery.objects.all()), LT25QuerySet)
+
+    def test_cte_create_manager_from_non_cteQuery(self):
+        class BrokenQuerySet(QuerySet):
+            "This should be a CTEQuerySet if we want this to work"
+
+        with self.assertRaises(TypeError):
+            class BrokenModel(Order):
+                class Meta:
+                    proxy = True
+                objects = CTEManager.from_queryset(BrokenQuerySet)()
```

Comment:
The class and manager instantiation seem unnecessary
```suggestion
            CTEManager.from_queryset(BrokenQuerySet)
```